plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java

```diff
@@ -271,42 +289,66 @@ public <T> T getAdapter(Class<T> adapter) {
     @Override
     public void primeEditorValue(@NotNull DBRProgressMonitor monitor, @NotNull StyledText control, @Nullable DBDContent value) throws DBException
     {
-        try {
-            // Load contents in two steps (empty + real in async mode). Workaround for some strange bug in StyledText in E4.13 (#6701)
-            final TextViewer textViewer = editor.getTextViewer();
-            final String encoding = getPanelSettings().get(PREF_TEXT_EDITOR_ENCODING);
-            final ContentEditorInput textInput = new ContentEditorInput(valueController, null, null, encoding, monitor);
-            boolean longContent = textInput.getContentLength() > LONG_CONTENT_LENGTH;
-            if (longContent) {
-                UIUtils.asyncExec(() -> {
-                    editor.setInput(new StringEditorInput("Empty", "", true, StandardCharsets.UTF_8.name()));
-                });
-            }
-            UIUtils.asyncExec(() -> {
-
-                if (textViewer != null && editor != null) {
-                    StyledText textWidget = textViewer.getTextWidget();
-                    if (textWidget != null && longContent) {
-                        GC gc = new GC(textWidget);
-                        try {
-                            UIUtils.drawMessageOverControl(textWidget, gc, NLS.bind(ResultSetMessages.panel_editor_text_loading_placeholder_label, textInput.getContentLength()), 0);
-                            editor.setInput(textInput);
-                        } finally {
-                            gc.dispose();
-                        }
-                    } else {
-                        editor.setInput(textInput);
-                    }
-                    applyEditorStyle();
-                }
-            });
-        } catch (Exception e) {
-            throw new DBException("Error loading text value", e);
-        } finally {
-            monitor.done();
-        }
+    	try {
+			int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore().getInt(ModelPreferences.EDITING_CONTENT_MAX_SIZE_KBYTES)*1000;
+			if (editor == null) {
+				log.error("Editor is null or undefined");
+				return;
+			}
+			resetEditorInput();
+			byte[] bytes = CommonUtils.toString(value).getBytes();
```

Comment:
Why don't you use getContentLength method?

Reply:
The main reason is in next: funtion "getContentLength" packed to ContentEditorInput here such object is unaccessable, (another logic branch is for the case of limited content Im using StringEditorInput object) that is why using getBytes is applicable here.

Reply:
I referred to `DBDContent`'s method as value is a `DBDContent`